pkg/cmd/pr/view/view.go

```diff
@@ -405,3 +393,46 @@ func prStateWithDraft(pr *api.PullRequest) string {
 
 	return pr.State
 }
+
+func retrievePullRequest(opts *ViewOptions) (*api.PullRequest, error) {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return nil, err
+	}
+
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	pr, repo, err := shared.PRFromArgs(apiClient, opts.BaseRepo, opts.Branch, opts.Remotes, opts.SelectorArg)
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.BrowserMode {
+		return pr, nil
+	}
+
+	errc := make(chan error)
+	count := 1
+	go func() {
+		reviews, err := api.ReviewsForPullRequest(apiClient, repo, pr)
+		pr.Reviews = *reviews
+		errc <- err
+	}()
+
+	if opts.Comments {
+		count++
+		go func() {
+			comments, err := api.CommentsForPullRequest(apiClient, repo, pr)
+			pr.Comments = *comments
+			errc <- err
+		}()
+	}
+
+	for i := 0; i < count; i++ {
+		if e := <-errc; e != nil {
+			err = e
+		}
+	}
```

Comment:
This could be replicated using `sync.WaitGroup`.
```golang
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		var reviews *api.PullRequestReviews
		reviews, err = api.ReviewsForPullRequest(apiClient, repo, pr)
		pr.Reviews = *reviews
	}()

	if opts.Comments {
		wg.Add(1)
		go func() {
			defer wg.Done()
			var comments *api.Comments
			comments, err = api.CommentsForPullRequest(apiClient, repo, pr)
			pr.Comments = *comments
		}()
	}

	wg.Wait()

	return pr, err
```
Do we have a preference on implementation?